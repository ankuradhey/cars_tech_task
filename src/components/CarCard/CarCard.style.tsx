import styled from "styled-components";

export const CardWrap = styled.div`
    display: flex;
    padding: ${(props) => props.theme.gutterSize.md};
`;

export const ImgWrap = styled.div`
    display: flex-row;
`;

export const ContentWrap = styled.div`
    display: flex-row;
    padding-left: ${(props) => props.theme.gutterSize.sm};
`;
