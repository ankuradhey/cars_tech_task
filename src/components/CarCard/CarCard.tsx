import React, { FC } from "react";
import { Card } from "react-bootstrap";
import { useHistory } from "react-router-dom";
import { CarCardProps } from "../../types";
import { ImgCard } from "../ImgCard";
import { Link } from "../Link";
import { Text } from "../Text";
import * as Styled from "./CarCard.style";

export const CarCard: FC<CarCardProps> = ({ id, pictureUrl, title, subTitle }) => {
    const history = useHistory();

    const navigateTo = () => {
        history.push(`/car/${id}`);
    };

    return (
        <Card className="margin-bottom-2">
            <Styled.CardWrap>
                <Styled.ImgWrap>
                    <ImgCard width="100px" height="80px" src={pictureUrl} />
                </Styled.ImgWrap>
                <Styled.ContentWrap>
                    <Card.Title>{title}</Card.Title>
                    <Text size="regular">{subTitle}</Text>
                    <Link data-testid="viewDetailsLink" className="d-block" onClick={navigateTo}>
                        View details
                    </Link>
                </Styled.ContentWrap>
            </Styled.CardWrap>
        </Card>
    );
};
