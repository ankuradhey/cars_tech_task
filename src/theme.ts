import { DefaultTheme } from "styled-components";

export const defaultTheme: DefaultTheme = {
    borderRadius: "2px",
    palette: {
        common: {
            black: "#4A4A4A",
            white: "#EDEDED",
        },
        primary: {
            main: "#EA7F28",
            contrastText: "#ffffff",
        },
        secondary: {
            main: "#D37324",
            contrastText: "#ffffff",
        },
        tertiary: {
            main: "#ffffff",
            contrastText: "#EDEDED",
        },
    },
    gutterSize: {
        sm: "8px",
        md: "12px",
        lg: "24px",
    },
};
