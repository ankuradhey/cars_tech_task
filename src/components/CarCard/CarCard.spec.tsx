import React from "react";
import { fireEvent, render, act } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router, Route } from "react-router-dom";
import { CarCard } from "./CarCard";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../../theme";

describe("CarCard Component", () => {
    let renderComponent;
    const history = { ...createMemoryHistory(), push: jest.fn(() => {}) };

    beforeEach(() => {
        renderComponent = render(
            <ThemeProvider theme={defaultTheme}>
                <Router history={history}>
                    <Route path="/">
                        <CarCard
                            id="1"
                            pictureUrl="/image"
                            title="car title"
                            subTitle="car subtitle"
                        />
                    </Route>
                </Router>
            </ThemeProvider>
        );
    });

    it("Should render CarCard component", () => {
        const { container, getByText } = renderComponent;
        expect(getByText("car title")).toBeInTheDocument();
        expect(getByText("car subtitle")).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it("Should navigate to Car details page when view details link is clicked", () => {
        const { getByTestId } = renderComponent;
        act(() => {
            fireEvent.click(
                getByTestId("viewDetailsLink"),
                new MouseEvent("click", {
                    bubbles: true,
                    cancelable: true,
                })
            );
        });
        expect(history.push).toHaveBeenCalledWith("/car/1");
    });
});
