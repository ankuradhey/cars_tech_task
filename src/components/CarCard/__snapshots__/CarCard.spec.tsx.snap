// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarCard Component Should render CarCard component 1`] = `
<div
  class="margin-bottom-2 card"
>
  <div
    class="sc-hKgILt hZGhHr"
  >
    <div
      class="sc-eCssSg ituOTJ"
    >
      <div
        class="sc-bdfBwQ dQCTDP"
        height="80px"
        loading="false"
        width="100px"
      >
        <img
          alt="car"
          src="/image"
        />
      </div>
    </div>
    <div
      class="sc-jSgupP gstVBl"
    >
      <div
        class="card-title h5"
      >
        car title
      </div>
      <div
        class="sc-dlfnbm dXPfWA"
      >
        car subtitle
      </div>
      <button
        class="sc-gsTCUz ezYRfg d-block"
        data-testid="viewDetailsLink"
      >
        View details
      </button>
    </div>
  </div>
</div>
`;
